### DISCUSSION TITLE
Filtering and diagrams based on a formula

### DISCUSSION THREAD

**Discussion by Sunsetpff** [Budibase/budibase#11501] [Help]
I am making an app that tracks expiration dates of workers licences so i made a table with worker name adress and licence expiration date and i need  one column that would track that expiration date and show if it is valid, expired or soon to expire based on todays date so i wrote this formula 

```js
const expDate = new Date($("Expiration date"));
const currentDate = new Date();
const twoMonthsAfter = new Date();
twoMonthsAfter.setMonth(currentDate.getMonth() + 2);
if (expDate > twoMonthsAfter) {
  return "Valid";
} else if (expDate > currentDate) {
  return "Soon to expire";
} else {
  return "Expired";
}
```

Now i want to be able to filter all the expired or valid rows and show a diagram with number of expired or soon to expire licences but i don't know how to do that, I'm not able to sort, filter or search anything based on this formula.

**Comment by melohagan** [comment]
Hey @Sunsetpff 

If you want to filter on a formula column, it must be a **static** formula.
![Screenshot 2023-08-14 at 09 04 05](https://github.com/Budibase/budibase/assets/101575380/d422601d-e980-4a1a-bb67-2c9bb4173047)
